### PR TITLE
Admin guard

### PR DESCRIPTION
**Pull request by robriks**
- Implemented AdminGuard.sol
- modernized repo dependencies (OZ-contracts)
- gitignore foundry & hardhat generated cache/out files
- silence compiler warnings on test files

### CODE REVIEW

File: contracts/examples/guards/AdminGuard.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        );
+
+        stateHash = _hashSafeSensitiveState();
+    }
+
+    function _hashSafeSensitiveState() internal view returns (bytes32) {
+        address singleton = address(uint160(uint256(bytes32(StorageAccessible(msg.sender).getStorageAt(0, 1)))));
+        bytes32 fallbackHandlerSlot = keccak256("fallback_manager.handler.address");
+        address fallbackHandler = address(uint160(uint256(bytes32(StorageAccessible(msg.sender).getStorageAt(uint256(fallbackHandlerSlot), 1)))));
+        address guard = address(this);
```
Comment (ilikesymmetry) [inline contracts/examples/guards/AdminGuard.sol]: Don't we need to read the value of the guard from the `GUARD_STORAGE_SLOT`?

**Review by ilikesymmetry — APPROVED** [review]
lgtm except for the guard bit!

File: contracts/examples/guards/AdminGuard.sol [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        (address[] memory modules,) = ModuleManager(msg.sender).getModulesPaginated(address(0x1), type(uint256).max);
+        address[] memory owners = OwnerManager(msg.sender).getOwners();
+        uint256 ownerCount = owners.length;
```
Comment (ilikesymmetry) [inline contracts/examples/guards/AdminGuard.sol]: would prefer to read this via another call to the Safe for airtight dependency, but this invariant of ownerCount and the array length should be good as is

File: contracts/examples/guards/AdminGuard.sol (hunk 1 above)
Comment (ilikesymmetry) [reply contracts/examples/guards/AdminGuard.sol]: yeah I'm pretty sure as is, I can change the guard and this hash does not detect that

Comment (robriks) [reply contracts/examples/guards/AdminGuard.sol]: Great catch! Resolved in 852e00c

File: contracts/examples/guards/AdminGuard.sol (hunk 2 above)
Comment (robriks) [reply contracts/examples/guards/AdminGuard.sol]: Sure thing. Resolved in d02e57c

**Pull request merged**